adsmp/tasks.py

```diff
@@ -221,6 +221,11 @@ def reindex_records(bibcodes, force=False, update_solr=True, update_metrics=True
             # build the solr record
             if update_solr:
                 solr_payload = solr_updater.transform_json_record(r)
+                # ADS microservices assume the identifier field exists and contains the canonical bibcode:
+                if 'identifier' not in solr_payload:
```

Comment:
I just want to verify this because I'm not totally sure how solr_payload is structured: if solr_payload has individual keys for each nested field in bib_data, this should work.

Reply:
See https://github.com/adsabs/ADSPipelineMsg/blob/7b84a1db621df29f7bc1597a09c0433c166572ae/adsmsg/protobuf/master_pb2.py#L289